### DISCUSSION TITLE
coral + camera integrated with homeassitant

### DISCUSSION THREAD

**Discussion by heapxor** [blakeblackshear/frigate#3640] [General]
hello,
 i finally integrated coral + frigate within Homeassistant but i am confused... .

I have camera https://www.hikvision.com/en/products/IP-Products/Network-Cameras/Pro-Series-EasyIP-/DS-2CD2143G0-I-S-/
but as u can see it shows that camera fps is 5 only ... 

![image](https://user-images.githubusercontent.com/64676/184541290-71ff777c-10ad-49c3-bb03-b0099a2249bf.png)

also not sure if the recorded stream is really 2560x1440?

![image](https://user-images.githubusercontent.com/64676/184541353-0c94998f-f516-4859-a878-10b7d683f920.png)

is there a way to monitor how does google coral helps / and offload work from cpu? 

also thats what i can see inside HAAS dmesg.. not sure if thats issue or its okay?

[  355.554777] usb 2-2: new SuperSpeed USB device number 3 using xhci_hcd
[  355.568670] usb 2-2: New USB device found, idVendor=1a6e, idProduct=089a, bcdDevice= 1.00
[  355.568685] usb 2-2: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[ 1237.885403] usb 2-2: reset SuperSpeed USB device number 3 using xhci_hcd
[ 1239.624567] usb 2-2: reset SuperSpeed USB device number 3 using xhci_hcd
[ 1239.637849] usb 2-2: LPM exit latency is zeroed, disabling LPM.
[ 1239.637853] usb 2-2: device firmware changed
[ 1239.646514] usb 2-2: USB disconnect, device number 3
[ 1239.824794] usb 2-2: new SuperSpeed USB device number 4 using xhci_hcd
[ 1239.837860] usb 2-2: LPM exit latency is zeroed, disabling LPM.
[ 1239.838604] usb 2-2: New USB device found, idVendor=18d1, idProduct=9302, bcdDevice= 1.00
[ 1239.838607] usb 2-2: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[ 2036.022206] usb 2-2: reset SuperSpeed USB device number 4 using xhci_hcd
[ 2036.035762] usb 2-2: LPM exit latency is zeroed, disabling LPM.

thanks!

**Comment by NickM-27** [comment]
Please read the docs https://docs.frigate.video/guides/getting_started#step-3-add-a-minimal-camera-configuration

This is an incorrect setup, the resolution you set for `detect` is way too high (in 99% of cases) for the detection stream. This should definitely be a sub stream if you have the option and 5 fps is the default. 

The stream with the record role will be used for recordings and kept at native resolution and frame rate

**Reply by heapxor** [reply]
the camera supports following ...
![image](https://user-images.githubusercontent.com/64676/184541789-4489a710-c51d-40a8-bedf-e1134105b2a1.png)


so is there some best practice on what resolution its good idea to run detection with coral?

thanks

**Reply by NickM-27** [reply]
The coral does not care about resolution directly, all regions that are sent for object detection will be resized to 320x320. 

The main concern with the resolution is the size of objects you are looking to detect and also CPU usage to run motion detection. See this section of the docs for guidance on that: https://docs.frigate.video/guides/camera_setup#choosing-a-detect-resolution. In this case, I would say a good start would be using the `Third Stream` with 1280 x 720

**Comment by heapxor** [comment]
@NickM-27 in case its resized to 320x320 whats the point to use 1280x720? thanks!

**Reply by NickM-27** [reply]
Object detection is run on `regions` not the entire camera frame. So if motion happens at a certain part of the camera frame, a `region` is calculated and that is what is resized to 320x320 and sent for object detection. Basically meaning that higher resolutions will help detect smaller objects, but especially for a first setup you want to start smaller and see how well things work.

**Reply by NickM-27** [reply]
You can look at this screenshot I took of the http://demo.frigate.video/ where the green box is the `region` sent for object detection (and resized to 320x320), the red box is the motion box, and the orange boxes are the objects detected.

<img width="1226" alt="Screen Shot 2022-08-14 at 8 43 05 AM" src="https://user-images.githubusercontent.com/14866235/184542171-0c23393a-c577-4855-88da-fc6d56ca8e92.png">

you can see how the green `region` is not anywhere near the size of the entire camera frame, and yet a higher detection resolution would not have helped in this case
